x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -159,6 +176,16 @@ private void includeAndExcludeFields(Set<String> fields, String[] index) {
         }
     }
 
+    private void checkRequiredFieldsArePresent(Set<String> fields) {
+        Set<String> missingFields = config.getAnalysis().getRequiredFields()
+            .stream()
+            .filter(f -> fields.contains(f) == false)
+            .collect(Collectors.toSet());
```

Comment:
```suggestion
            .collect(Collectors.toList());
```
`List` seems more appropriate here as we are not concerned about duplicates. And, I think it might make the print out better?

Reply:
Fair enough. I also need to change the variable type though so I'll push a commit.